packages/next/pages/_document.js

```diff
@@ -10,7 +10,8 @@ const Fragment = React.Fragment || function Fragment ({ children }) {
 
 export default class Document extends Component {
   static childContextTypes = {
-    _documentProps: PropTypes.any
+    _documentProps: PropTypes.any,
+    _devOnlyInvalidateCacheQueryString: PropTypes.string,
```

Comment:
Should this use 16.3 context instead?

Reply:
The documentProps one is kinda tricky but your new one might make sense

Reply:
I will think of some way to refactor this. With the current implementation I think that we would need to add the provider in next-server as the Document `render` method could be overridden. 

After all probably your idea of adding the timestamp in next-server was better - we could define all the files to render there (also the hardcoded ones) and have this component just render them.